py-polars/polars/plugins.py

```diff
@@ -0,0 +1,37 @@
+from __future__ import annotations
+
+from pathlib import Path
+
+from polars._utils.unstable import unstable
+
+__all__ = ["get_shared_lib_location"]
+
+
+@unstable()
+def get_shared_lib_location(package_init_path: str) -> str:
+    """
+    Get location of Shared Object file.
+
+    See the `user guide <https://docs.pola.rs/user-guide/expressions/plugins/>`_
+    for more information about plugins.
+
+    .. warning::
+        This functionality is considered **unstable**. It may be changed
+        at any point without it being considered a breaking change.
+
+    Parameters
+    ----------
+    package_init_path
+        The ``__init__.py`` file of the plugin package.
+
+    Returns
+    -------
+    str
+        The location of the Shared Object file.
+    """
+    directory = Path(package_init_path).parent
```

Comment:
I don't think it makes a lot of sense for function to expect a path to the init file within the package. It makes more sense to accept any Path, and we will look for a shared lib in that path.

For ease of use, we can allow file inputs, in which case we'll look in the directory that file lives in (e.g. use `parent`). So then this will still support calling with `__file__` in the package init.

So this function should accept a `Path | str`, and we convert it to a Path, then check if it's a directory, if not, use `.parent`.

Reply:
Actually, isn't it just a lot cleaner if this functionality is just part of `register_plugin`? That the `lib` argument can accept a Path and it will find the library in that path? Then this whole util doesn't need to be public.